server.js

```diff
@@ -0,0 +1,138 @@
+import express from "express";
+import bodyParser from "body-parser";
+import mongoose, { ObjectId } from "mongoose";
+const app = express();
+const PORT = 3001;
+const connectionString =
+  "mongodb+srv://Prohar:PEA16april2005@cluster0.bosgtkc.mongodb.net/?retryWrites=true&w=majority";
+
+const URL = {
+  GREETINGS: "/api/greetings",
+  NOTES: "/api/notes",
+};
+
+app.use(express.json());
+
+// DB CONNECTION
+mongoose
+  .connect(connectionString, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+  })
+  .then(() => {
+    console.log("Connected to MongoDB");
+  })
+  .catch((error) => {
+    console.error("Error connecting to MongoDB:", error.message);
+  });
+
+// SCHEMA
+const schema = {
+  id: Number,
+  title: String,
+  content: String,
+  createAt: Date,
+  updateAt: Date,
+};
+const mongooseModel = mongoose.model("NEWCOL", schema);
+
+// TASK1
+app.get(URL.GREETINGS, function (req, res) {
+  try {
+    const name = req.query.name;
+    if (name) {
+      res.send(`hello ${name}`);
+    } else {
+      throw new Error("there is no value");
+    }
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+// GET
+app.get(URL.NOTES, async (req, res) => {
+  res.send({});
+});
+
+// POST
+app.post(URL.NOTES, async (req, res) => {
+  console.log("inside post function");
+
+  const data = new mongooseModel({
+    id: req.body.id,
+    title: req.body.title,
+    content: req.body.content,
+    createAt: Date.now(),
+    updateAt: "",
+  });
+
+  const val = await data.save();
+  res.send(data);
+});
+
+// PUT
+app.put(URL.NOTES + "/:id", async (req, res) => {
+  let updateId = req.params.id;
+  let updateTitle = req.body.title;
+  let updateContent = req.body.content;
+  let updateDate = Date.now();
+
+  mongooseModel
+    .findOneAndUpdate(
+      { id: updateId },
+      {
+        $set: {
+          title: updateTitle,
+          content: updateContent,
+          updateAt: updateDate,
+        },
+      },
+      { new: true }
+    )
+    .exec()
+    .then((data) => {
+      if (data === null) {
+        res.send("nothing found");
+      } else {
+        res.send(data);
+      }
+    })
+    .catch((err) => {
+      console.error("Error updating note:", err.message);
+      // Handle the error
+      res.status(500).send("Internal Server Error");
+    });
+});
```

Comment:
Async await!
Move all the parts of the method to the particular folder: routes, controllers, providers.
P.S. Once server gets a request from client or Postman the flow of server treatment: Routes (navigate to particular endpoint) -> Controller (process request, calculating data) -> Provider (working with DB requests) -> Back to the same Controller (preparing response or additional DB requests) -> Sending Response